app/validators/flood_risk_engine/grid_reference_validator.rb

```diff
@@ -0,0 +1,52 @@
+# Validate that a grid reference can be processed.
+module FloodRiskEngine
+  class GridReferenceValidator < ActiveModel::EachValidator
```

Comment:
not sure if there were patterns missed in WEX but this was what we used and passed QA at the time

GRID_REFERENCE_REGEX = /\A[a-zA-Z]{2}\d{6,10}\z/i